scripts/assemble_changelog.py

```diff
@@ -56,46 +56,116 @@ def __init__(self, filename, line):
         message = ('Lost content from {}: "{}"'.format(filename, line))
         super().__init__(message)
 
-STANDARD_SECTIONS = (
-    b'Interface changes',
+STANDARD_CATEGORIES = (
+    b'API changes',
     b'Default behavior changes',
     b'Requirement changes',
     b'New deprecations',
     b'Removals',
-    b'New features',
+    b'Features',
     b'Security',
-    b'Bug fixes',
-    b'Performance improvements',
-    b'Other changes',
+    b'Bugfix',
+    b'Changes',
 )
 
-class ChangeLog:
-    """An Mbed TLS changelog.
+CategoryContent = namedtuple('CategoryContent', [
+    'name', 'title_line', # Title text and line number of the title
+    'body', 'body_line', # Body text and starting line number of the body
+])
 
-    A changelog is a file in Markdown format. Each level 2 section title
-    starts a version, and versions are sorted in reverse chronological
-    order. Lines with a level 2 section title must start with '##'.
+class ChangelogFormat:
+    """Virtual class documenting how to write a changelog format class."""
 
-    Within a version, there are multiple sections, each devoted to a kind
-    of change: bug fix, feature request, etc. Section titles should match
-    entries in STANDARD_SECTIONS exactly.
+    @classmethod
+    def extract_top_version(cls, changelog_file_content):
+        """Split out the top version section.
 
-    Within each section, each separate change should be on a line starting
-    with a '*' bullet. There may be blank lines surrounding titles, but
-    there should not be any blank line inside a section.
-    """
+        Return ``(header, top_version_title, top_version_body, trailer)``
+        where ``changelog_file_content == header + top_version_title +
+                                          top_version_body + trailer``.
```

Comment:
I think with the next commit (that creates a new section if the current one is a released version) changes the meaning of that identity a bit, perhaps the documentation should mention it.

Reply:
The identity is still true for the new content when the function return. Maybe I should rename the function to `extract_or_create_top_version`?

Reply:
I agree the identity still holds, but from just reading the description it's not clear that it might be for a new value of `changelog_file_content`. I think renaming the function would make it clearer indeed.